mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageMetaData.java

```diff
@@ -24,31 +24,73 @@
 
 import org.apache.james.mailbox.MessageUid;
 
-public interface MessageMetaData {
+import com.google.common.base.Objects;
 
-    MessageUid getUid();
-    
-    
-    /**
-     * Return the modify-sequence number of the message. This is kind of optional and the mailbox
-     * implementation may not support this. If so it will return -1
-     */
-    long getModSeq();
+public class MessageMetaData {
+    private final MessageUid uid;
+    private final Flags flags;
+    private final long size;
+    private final Date internalDate;
+    private final long modSeq;
+    private final MessageId messageId;
+
+    public MessageMetaData(MessageUid uid, long modSeq, Flags flags, long size, Date internalDate, MessageId messageId) {
+        this.uid = uid;
+        this.flags = flags;
+        this.size = size;
+        this.modSeq = modSeq;
+        this.internalDate = internalDate;
+        this.messageId = messageId;
+    }
+
+    public Flags getFlags() {
+        return flags;
+    }
 
-    Flags getFlags();
-    
     /**
      * Return the size in bytes
      */
-    long getSize();
+    public long getSize() {
+        return size;
+    }
 
     /**
      * <p>
      * IMAP defines this as the time when the message has arrived to the server
      * (by smtp). Clients are also allowed to set the internalDate on append.
      * </p>
      */
-    Date getInternalDate();
-    
-    MessageId getMessageId();
+    public Date getInternalDate() {
+        return internalDate;
+    }
+
+    public MessageUid getUid() {
+        return uid;
+    }
+
+    public MessageId getMessageId() {
+        return messageId;
+    }
+
+    /**
+     * Return the modify-sequence number of the message. This is kind of optional and the mailbox
+     * implementation may not support this. If so it will return -1
+     */
+    public long getModSeq() {
+        return modSeq;
+    }
+
+    @Override
+    public boolean equals(Object obj) {

```

Comment:
can we trust these `equals()` and `hashCode()`? I see they use only `uid`, what about other properties?

Reply:
I did not intend to change that logic to be honest ^^